src/data/validate.rs

```diff
@@ -0,0 +1,365 @@
+use crate::config::Mapping;
+
+/// Validate paths for entity
+pub fn validate_paths_for_entity(
+    entity: &str,
+    mappings: &Vec<Mapping>,
+    api_schema: &serde_json::Map<String, serde_json::Value>,
+) -> anyhow::Result<()> {
+    // if entity name is not set in api_schema throw an exception
+    if !api_schema.contains_key(entity) {
+        anyhow::bail!("Entity {} not found in API schema", entity);
+    }
+
+    for entry in mappings {
+        let path_mapping = match entry {
+            Mapping::ByPath(path_mapping) => path_mapping,
+            Mapping::ByScript(_) => continue,
+        };
+
+        let path = path_mapping.entity_path.split('.').collect::<Vec<_>>();
+
+        // if path starts with ? its optional
+        if path[0].ends_with('?') {
+            continue;
```

Comment:
I don't know if this really makes sense. Sure you could argue that `manufacturer?.media?.url` makes the `url` optional in this case, but in general the `?` references the association before it (and allows to return early if it is not there).

So what about: `tax.otherOptionalAssociation?.field`: in this case `tax` is required and always there but other associations later down the chain might not be 🤔

Reply:
Also, you still want to validate the path anyways right? e.g. `manufacturer?.media?.url` has `manufacturer` as an association and that one has a `media` association and that has the field `url`